### ISSUE TITLE
When second click to selected node on QTree selected key becomes null

### ISSUE THREAD

**Issue by ryhn** [quasarframework/quasar#7701]
Hi,
I use selected.sync and expanded.sync prop. After second click to selected node than selectedKey value becomes null. I think selected-key must stay the same.

**Comment by rstoenescu** [comment]
Please supply a codepen as requested...

**Comment by rstoenescu** [comment]
Please supply the requested codepen. Most likely it's something wrongly understood about QTree, but we can't help if we don't get to see your code.

**Comment by ryhn** [comment]
Hi, codepen sample a lot of works for me :) 

Here my QTree and @update event code

```
<q-tree
        ref="menuTree"
        :nodes="mainMenu"
        node-key="id"
        :selected.sync="selectedKey"
        :expanded.sync="expanded"
        @update:selected="goTo" 
/>
```
```
 goTo(val) {
      let x = this.$refs.menuTree.getNodeByKey(val);
      console.log(val);
      if (!x) return;
      if (!x.to) {
        let isexpanded = this.expanded.indexOf(x.id) > -1;
        this.$refs.menuTree.setExpanded(x.id, !isexpanded);
        return;
      }
      
      this.$router.push({ path: x.to });
    }
```
I am trying to toggle child menu items to click parent menu node. In goTo(val) function "val" value changed to "null" on second click. So for toggle action, it needs to be clicked three times to parent node for toggle child menu.

Without :selected.sync prop QTree toggle child nodes it works good. With :selected.sync QTree child nodes are not clickable

All of these codes are to achieve toggle menu actions. :)

**Comment by super-seo** [comment]
Hi.
Example
1.https://codepen.io/superseo/pen/XWeyBpz
2.https://codepen.io/superseo/pen/oNGQMQO?editors=101

Taken from the manual https://quasar.dev/vue-components/tree

The second click on the selected element always resets the selection. That is, click on the element for the first time - the corresponding Tab Panel opens, click again on the same element - it closes.
But it is necessary that the second click on the same element does not cause deselection.

Adding an extra props "unSelectable" is probably enough. By default (true) - old behavior. If false, then the second click does not deselect. For those cases when such behavior is necessary, without breaking the old logic.

https://codepen.io/superseo/pen/xxXQJLz?editors=101
* added a new parameter, for example on PR **https://github.com/quasarframework/quasar/pull/12012**
* just use <q-tree ... :un-selectable="false" />

The error from the description occurs because of a check on this line
https://github.com/pdanpdan/quasar/blob/8a5a2d799bc4af2dc185ca0ec1614b7dd66737aa/ui/src/components/tree/QTree.js#L607

Same request but for v1
https://github.com/quasarframework/quasar/issues/3551

**Comment by rstoenescu** [comment]
New prop (no-selection-unset) will be available in Quasar v2.4.10 & v1.17.5
